imap_processing/cli.py

```
@@ -1206,6 +1207,80 @@ def do_processing(  # noqa: PLR0912
+            return super().post_processing(processed_data, dependencies)
+
+        ancillary_identifiers = [
+            "imap_mag_l1d_gradiometry-offsets-burst",
+            "imap_mag_l1d_gradiometry-offsets-norm",
+            "imap_mag_l1d_spin-offsets",
+        ]
+
+        ancillary_files = []
+        science_datasets = []
+        for dataset in processed_data:
+            logical_source = dataset.attrs["Logical_source"]
+            if logical_source in ancillary_identifiers:
+                # Skip write_cdf
+                instrument, _data_level, descriptor = dataset.attrs[
+                    "Logical_source"
+                ].split("_")[1:]
+                start_date = self.start_date
+                version = self.version
+
+                output_filename = (
+                    imap_data_access.AncillaryFilePath.generate_from_inputs(
+                        instrument=instrument,
+                        descriptor=descriptor,
+                        version=version,
+                        extension="cdf",
+                        start_time=start_date,
+                        end_time=start_date,
+                    ).filename
+                )
+
+                try:
+                    # write file to CDF
+                    xarray_to_cdf(
+                        dataset, output_filename, terminate_on_warning=False, istp=False
+                    )
+                    ancillary_files.append(output_filename)
+                except Exception as e:
+                    # Don't fail for any reason for ancillary files
+                    logger.warning(f"Hit error {e} when creating {output_filename}")
+                    continue
+            else:
+                science_datasets.append(dataset)
+
+        try:
+            self.upload_products(ancillary_files)
+        except Exception as e:
+            logger.warning(f"Failed to upload ancillary products due to error {e}")
```

If these are output products don't we think they should be done with the same care as science products? I'm not sure why these would fail when a science product wouldn't.

---

Well, they are currently failing and blocking science product release. Since these are internal watchdog products, I do think they are a lower priority than the science files... but I can remove the excepts if that doesn't line up with the rest of the SDC or what MAG wants.

---

I'm just curious if there is a reason they are failing. Can we identify this earlier in the process somehow and not include these in the list of data products to upload if there is bad content. i.e. the upload isn't what should fail and if it does we should fix that at the source.

---

That is what the test I added is checking! But, partly the problem is that they are trying to get uploaded as science files and they aren't. So, I don't anticipate this happening again, but if it does, I was thinking it would be preferable to NOT block science products. However, this does reduce visibility into these products if they start failing so I understand your concern. 
Really, these checks aren't for bad content, they're for A. is the filename constructed properly and B. is the ancillary file upload functioning properly. 

But I am definitely open to removing them if the increased stability isn't worth the risk of missing errors from the SDC perspective.

---

In the `post_processing()` method there is a callout for Path vs Dataset difference in the list of products. So my personal preference would be to write the ancillary products here as you have (logging/ignoring errors if there are any if you want in that portion), then call super().post_process() with the mixed list of all science datasets and ancillary filepaths, and let the uploads happen there.

---

Oh yeah, that's a good catch that makes this cleaner as well. I'll make that change.